tests/validation/tests/v3_api/test_bkp_restore.py

```diff
@@ -55,17 +84,23 @@ def test_bkp_restore():
                      skipIngresscheck=False)
 
     # Verify the ingress created before taking the snapshot
-    validate_ingress(namespace["p_client"], namespace["cluster"],
-                     [workload], host, path)
+    validate_ingress(p_client, cluster, [workload], host, path)
 
     # Verify the workload created after getting a snapshot does not exist
     # after restore
     workload_list = p_client.list_workload(uuid=testworkload.uuid).data
     print(len(workload_list))
     assert len(workload_list) == 0
 
+    if backup_mode == "s3":
+        # Check the backup reference is deleted in Rancher and S3
+        client.delete(cluster.etcdBackups(id=backupId)['data'][0])
+        wait_for_backup_to_delete(cluster, backupname)
+        assert AmazonWebServices().s3_backup_check(backupfilename) is False
+    cluster_cleanup(client, cluster)
```

Comment:
Can we have the "cluster_cleanup(client, cluster)" in a finalizer in the method where the cluster is created?

Reply:
I've moved the cleanup_cluster to the fixtures finalizers. It works I did testing `RANCHER_CLEANUP_CLUSTER=True` and `False. @sowmyav27